reconcile/utils/openshift_resource.py

```diff
@@ -326,6 +358,12 @@ def canonicalize(body):
         # remove status
         body.pop("status", None)
 
+        # remove controller managed labels
+        labels = body["metadata"].get("labels", {})
+        for label in set(labels.keys()):
+            if OpenshiftResource.is_controller_managed_label(body["kind"], label):
+                labels.pop(label)
```

Comment:
this means the controlled label has been set in app-interface right ? Should we log a kind of warning or even fail ?

Reply:
or is that the 'current' resource ?

Reply:
a label that is a `is_controller_managed_label` is one that is completely managed (existence and/or value) by something other then a-i. e.g. the `clusterId` label on a `ManagedCluster`. 

those labels should not be defined in a-i. we could also fail a PR check if such a label is present.

by removing those labels from the state (mostly relevant for current state), resources will still be considered equal in the sense of `__eq__` and `sha256sum()`